cmd/server-mux.go

```diff
@@ -43,22 +43,17 @@ var defaultHTTP1Methods = []string{
 	"CONNECT",
 }
 
-// ConnBuf - contains network buffer to record data
-type ConnBuf struct {
-	buffer []byte
-	unRead bool
-	offset int
-}
-
 // ConnMux - implements a Read() which streams twice the firs bytes from
 // the incoming connection, to help peeking protocol
 type ConnMux struct {
 	net.Conn
 	lastError error
-	dataBuf   ConnBuf
+	readLen   int
+	bufwr     *bufio.ReadWriter
 }
 
-func longestWord(strings []string) int {
+// Find the longest word of all the HTTP methods.
+func longestWord() int {
 	maxLen := 0
 	for _, m := range defaultHTTP1Methods {
```

Comment:
instead of having 2 loops one for defaultHTTP1Methods and one for defaultHTTP2Methods, this can be done:

`for _, m := range append(defaultHTTP1Methods, defaultHTTP2Methods) {...}`